client/app/components/dashboards/CreateDashboardDialog.jsx

```diff
@@ -0,0 +1,86 @@
+import { trim } from 'lodash';
+import React, { useRef, useState, useEffect } from 'react';
+import Modal from 'antd/lib/modal';
+import Input from 'antd/lib/input';
+import DynamicComponent from '@/components/DynamicComponent';
+import { wrap as wrapDialog, DialogPropType } from '@/components/DialogWrapper';
+import { $location, $http } from '@/services/ng';
+import recordEvent from '@/services/recordEvent';
+import { policy } from '@/services/policy';
+
+function CreateDashboardDialog({ dialog }) {
+  const [name, setName] = useState('');
+  const [isValid, setIsValid] = useState(false);
+  const [saveInProgress, setSaveInProgress] = useState(false);
+  const inputRef = useRef();
+  const isCreateDashboardEnabled = policy.isCreateDashboardEnabled();
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      if (inputRef.current) {
+        inputRef.current.focus();
+      }
+    }, 100);
+    return () => clearTimeout(timer);
+  }, []);
+
+  function handleNameChange(event) {
+    const value = trim(event.target.value);
+    setName(value);
+    setIsValid(value !== '');
+  }
+
+  function save() {
+    if (name !== '') {
+      setSaveInProgress(true);
+
+      $http.post('api/dashboards', { name })
```

Comment:
Wdyt of `_.capitalize(name)`?

Reply:
Let's keep it as is (at least, for now). Automatic capitalization is not so brilliant idea in most cases (just because algorithm is too generic), especially if we do it on save - so user will have no chance to fix typos introduced by `_.capitalize` (I think that's bad UX?)